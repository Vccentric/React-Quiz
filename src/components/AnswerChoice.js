import React from "react";
import PropTypes from "prop-types";

// answer choice component
const AnswerChoice = ({ value, checkAnswer }) => {
  return (
    <button className="answer" onClick={checkAnswer}>
      {value}
    </button>
  );
};

AnswerChoice.propTypes = {
  value: PropTypes.string,
  checkAnswer: PropTypes.func,
};

export default AnswerChoice;
