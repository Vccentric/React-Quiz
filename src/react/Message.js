import React from 'react';
import { hot } from 'react-hot-loader';

// message component
function Message(props) {
    const { isAnswered, isCorrect } = props;
    return (
        <div
            id="message"
            style={isAnswered ? { display: 'block' } : { display: 'none' }}
            className={isCorrect ? 'correct' : 'wrong'}
        >
            {isCorrect ? 'Correct Answer!' : 'Wrong Answer!'}
        </div>
    );
}

export default hot(module)(Message);
