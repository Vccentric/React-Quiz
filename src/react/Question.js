import React from 'react';
import { hot } from 'react-hot-loader';

// question component
function Question(props) {
    const { question } = props;
    return <p id="question">{question}</p>;
}

export default hot(module)(Question);
