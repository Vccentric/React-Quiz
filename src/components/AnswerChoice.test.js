import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import React from "react";
import AnswerChoice from "./AnswerChoice";

// setup variables
const value = "test";
const selectAnswer = jest.fn();

test("renders correctly - snapshot", () => {
  const tree = renderer
    .create(
      <AnswerChoice
        value={value}
        selected={false}
        selectAnswer={selectAnswer}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders answer choice as selected", () => {
  const { container } = render(
    <AnswerChoice value={value} selected={true} selectAnswer={selectAnswer} />
  );
  const element = container.getElementsByClassName("selected");
  expect(element?.length).toBe(1);
});
