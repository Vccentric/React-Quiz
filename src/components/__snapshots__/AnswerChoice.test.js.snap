// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - snapshot 1`] = `
<button
  className="answer"
  onClick={[MockFunction]}
>
  test
</button>
`;
